### PR TITLE
Wrote documentation for aperture catalog generation

### PR DESCRIPTION
**Pull request by dulude**
- See JIRA ticket [HLA-356](https://jira.stsci.edu/browse/HLA-356).
- Created doc/source/catalog_generation.rst, which contains documentation for just the aperture portion of catalog generation.
- modified doc/source/index.rst so that a link to the catalog_generation.html page would be created in index.html.

### CODE REVIEW

File: doc/source/catalog_generation.rst [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+1: Source Detection
+-------------------
+
+1.1: Important Clarifications
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+As described in the previous step, AstroDrizzle creates a single multi-filter detector-level drizzle-combined
+image for source identification and one or more detector/filter-level drizzle-combined images (depending on
+which filters were used in the dataset) for photometry. The same set of sources identified in the
+multi-filter detection image is used to measure photometry for each filter. We use method to maximize the
+signal across all available wavelengths at the source detection stage, thus providing photometry with the
```
Comment (mdlpstsci) [inline doc/source/catalog_generation.rst:19]: Missing work in parentheses...We use (this) method to...

File: doc/source/catalog_generation.rst [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+
+1.2.2: Detection Image Background Subtraction
+""""""""""""""""""""""""""""""""""""""""""""""""
+To ensure optimal source detection, the multi-filter detection image is background-subtracted. We computed a
+2-dimensional background image using the `photutils.background.Background2d <https://photutils.readthedocs.io/en/stable/api/photutils.background.Background2D.html>`_ Astropy tool. This algorithm uses
+sigma-clipped statistics to determine background and RMS values across the image. An initial low-resolution
+estimate of the background is performed by computing sigma-clipped median values in 27x27 pixel boxes across
+the image. This low-resolution background image is then median-filtered using a 3x3 pixel sample window to
+correct for local small-scale overestimates and/or underestimates.
+
```
Comment (mdlpstsci) [inline doc/source/catalog_generation.rst:48]: For the 27x27 and 3x3, it should be noted these are configurable values.  Our catalogs use these values deeming them to be the best for the general situation, but a user could twiddle the values to optimize for their own data.

File: doc/source/catalog_generation.rst [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+.. math::
+    \Delta ABmag_{outer} \leq  \frac{2.5}{snr \cdot log(10))}
+
+Where
+    * :math:`{\Delta ABmag_{outer}}` is the outer aperture AB magnitude uncertainty
+    * :math:`{snr}` is the signal to noise ratio, which is 1.5 for ACS/WFC and 5.0 for all other detectors.
+
+2.3.2.4: Assignment of flag value 32 (false detection: swarm around saturated source)
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+The source identification routine has been shown to falsely identify sources in regions near bright or saturated
```
Comment (mdlpstsci) [inline doc/source/catalog_generation.rst:258]: to identify falsely

File: doc/source/catalog_generation.rst [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+appropriately.
+
+3: The output catalog file
+---------------------------
+3.1: Filename format
+^^^^^^^^^^^^^^^^^^^^^^
+Source positions and photometric information are written to a .ecsv (Enhanced Character Separated Values) file. The
+naming of this file is fully automatic and follows the following format:
+<TELESCOPE>_<PROPOSAL ID>_<VISIT NUMBER>_<INSTRUEMENT>_<DETECTOR>_
+<FILTER>_<DATASET NAME>_<CATALOG TYPE>.ecsv
```
Comment (mdlpstsci) [inline doc/source/catalog_generation.rst:283]: Please correct spelling of INSTRUMENT.

File: doc/source/catalog_generation.rst [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+
+So, for example if we have the following information:
+    * Telescope = HST
+    * Proposal ID = 98765
+    * Visit Number = 43
+    * Instrument = acs
```
Comment (mdlpstsci) [inline doc/source/catalog_generation.rst:289]: Same comment as above for "visit" vs "obset".

File: doc/source/catalog_generation.rst [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+    * MagErrAp2: Outer aperture brightness uncertainty, in AB magnitude
+    * MSkyAp2: Outer aperture background brightness, in AB magnitude
+    * StdevAp2: Standard deviation of the outer aperture background brightness, in AB magnitude
+    * FluxAp2: Outer aperture flux, in electrons/sec
+    * CI: Concentration index (MagAp1 – MagAp2), in AB magnitude
+    * Flags: See Section 2.3.2 for flag value definitions
+
+Segment Photometric Catalog Generation
+=======================================
+Michele's documentation goes here!
```
Comment (mdlpstsci) [inline doc/source/catalog_generation.rst:361]: Yeah, yeah, yeah.  Fortunately for some items, I can just refer to your docs!

File: doc/source/catalog_generation.rst (hunk 1 above)
Comment (dulude) [reply doc/source/catalog_generation.rst:19]: Issue resolved in commit [203ed51](https://github.com/spacetelescope/drizzlepac/pull/760/commits/203ed519b49749f7ad519df95710174ab98faffe)

File: doc/source/catalog_generation.rst (hunk 2 above)
Comment (dulude) [reply doc/source/catalog_generation.rst:50]: Issue resolved in commit [203ed51](https://github.com/spacetelescope/drizzlepac/pull/760/commits/203ed519b49749f7ad519df95710174ab98faffe)

File: doc/source/catalog_generation.rst (hunk 3 above)
Comment (dulude) [reply doc/source/catalog_generation.rst]: Issue resolved in commit [203ed51](https://github.com/spacetelescope/drizzlepac/pull/760/commits/203ed519b49749f7ad519df95710174ab98faffe)

File: doc/source/catalog_generation.rst (hunk 4 above)
Comment (dulude) [reply doc/source/catalog_generation.rst:285]: Fixed misspelling of "INSTRUMENT" in commit [44187d9](https://github.com/spacetelescope/drizzlepac/pull/760/commits/44187d9376ea9158415c99c2f42cd63ca72563e8)

Comment (dulude) [reply doc/source/catalog_generation.rst:285]: "Visit number" changed to "Observation set ID" in commit [2e21c00](https://github.com/spacetelescope/drizzlepac/pull/760/commits/2e21c0043d31e857b7478b34e33362fc56392baf)

File: doc/source/catalog_generation.rst (hunk 5 above)
Comment (dulude) [reply doc/source/catalog_generation.rst:291]: "Visit number" changed to "Observation set ID" in commit [2e21c00](https://github.com/spacetelescope/drizzlepac/pull/760/commits/2e21c0043d31e857b7478b34e33362fc56392baf)

**Pull request merged**